src/service/resource.js

```diff
@@ -181,6 +181,9 @@ export class Resource {
     /** @private {?Function} */
     this.renderOutsideViewportResolve_ = null;
 
+    /** @private {number} */
+    this.renderOutsideViewportMultiplier_ = 1;
```

Comment:
I do want this to prematurely trigger layoutCallback.  I suppose I could add logic to renderOutsideViewport that checks if adPromise has been populated.  Would that be safe?